src/services/checkpoints/CheckpointService.ts

```diff
@@ -291,7 +298,7 @@ export class CheckpointService {
 			// the checkpoint (i.e. the `git restore` command doesn't work
 			// for empty commits).
 			await fs.writeFile(path.join(baseDir, ".gitkeep"), "")
-			await git.add(".")
+			await git.add(".gitkeep")
```

Comment:
I encountered this subtle bug:

In the case where we initialize a git repo we end up committing the first change that Roo Code makes in the initial commit, which prevents us from creating a corresponding checkpoint for it. To address this I only commit the `.gitkeep` file (which we create) on repo initialization.